### PR TITLE
Track cmi dependencies correctly

### PR DESCRIPTION
**Pull request by anmonteiro**
- make `.cmi` depend on their dependencies correctly
  - we were previously doing this for `.cmj` files, this change applies the same arguments to the `.cmi` module builds

Unfortunately writing interfaces is not that common in the ReScript world, which made this bug stay hidden for longer than I would have liked.

cc @jchavarri

### CODE REVIEW

File: jscomp/bsb/bsb_ninja_file_groups.ml [hunk 1]
```diff
@@ -126,6 +155,8 @@ let emit_module_build
       ~outputs:[output_cmi]
       ~inputs:[output_iast]
       ~rule:(if is_dev then rules.mi_dev else rules.mi)
+      ~bs_dependencies
+      ~rel_deps:(rel_bs_config_json :: relative_ns_cmi)
```
Comment (anmonteiro) [inline jscomp/bsb/bsb_ninja_file_groups.ml:159]: These are the only relevant changes here. Everything else is just moving code so that it's in scope.

Comment (EduardoRFS) [reply jscomp/bsb/bsb_ninja_file_groups.ml:159]: When doing this at least makes it into two commits, so that it can be reviewed separatedly.

**Review by EduardoRFS — APPROVED** [review]
> which made this bug stay hidden for longer than I would have liked.

+1, we need some big project to integrate on the CI on the future.

sad anime girl

**Comment by Mathspy** [comment]
> +1, we need some big project to integrate on the CI on the future.

Have you ever seen this https://github.com/rust-lang/crater? Here’s a recently example of it running https://github.com/rust-lang/rust/issues/80667#issuecomment-756597928 👀 
I’d honestly love for us to work on something like this, and possibly run only on a subset until we’re confident

**Pull request merged**